src/Util/Network.php

```diff
@@ -658,4 +658,25 @@ public static function isValidHttpUrl(string $url): bool
 		$scheme = parse_url($url, PHP_URL_SCHEME);
 		return !empty($scheme) && in_array($scheme, ['http', 'https']) && parse_url($url, PHP_URL_HOST);
 	}
+
+	/**
+	 * Check if a provided URI is valid
+	 *
+	 * @param string|null $uri
+	 * @return boolean
+	 */
+	public static function isValidUri(string $uri = null): bool
```

Comment:
So far I've been naming the methods that deal with string URI with `Url` and the one that deal with `UriInterface` object with `Uri`. See `isUrlBlocked` vs `isUriBlocked` for example.

Beyond that, since you are trying to instantiate an `Uri` object in this method, you could return it directly or null instead like this:
```php
try {
			return new Uri($uri);
		} catch (\Exception $e) {
			Logger::debug('Invalid URI', ['code' => $e->getCode(), 'message' => $e->getMessage(), 'uri' => $uri]);
			return null;
		}
```
Since it would be a factory, it probably should be renamed to something like `createUriFromUrl`.
```suggestion
	public static function createUriFromUrl(string $uri = null): ?UriInterface
```

Reply:
We can't name it `url`, because it can not only be an url that we provide here.

Reply:
`createUriFromString`?

Reply:
I'm unsure if this a suggestion before you saw that I named it that way or that you are questioning my naming choice.

Reply:
It was a naive suggestion, I'm glad we're on the same page!